### PR TITLE
Pin versions. Update deployment process.

### PR DESCRIPTION
**Pull request by chdsbd**
Pinned versions for Python and updated the deployment process to use
some bash files

- [x] Make Pipfile stuff work like package.json

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,12 @@ docker-compose -f docker-compose-dev.yml run -d --entrypoint "tail -f /dev/null"
 docker-compose -f docker-compose-dev.yml exec django sh boostrap-dev.sh
 ```
 
+### Updating dependencies
+#### Python
+After changing the Pipfile, you must do the following rebuild the django container and bring up docker compose so the new Pipfile can be copied from the container (`docker-compose -f docker-compose-dev.yml up --build django`).
```
Comment (sbdchd) [inline README.md]: Read that again

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
```
Comment (sbdchd) [inline build.sh]: Don’t we want something like set -ex in these scripts?

File: deploy.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
```
Comment (sbdchd) [inline deploy.sh]: Same as above

File: upload.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
```
Comment (sbdchd) [inline upload.sh]: Same as above

**Comment by chdsbd** [comment]
@sbdchd Updated

File: build [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
```
Comment (sbdchd) [inline build]: `#!/usr/bin/env sh`

File: deploy [hunk 6]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
```
Comment (sbdchd) [inline deploy]: `#!/usr/bin/env sh`

File: frontend/bootstrap.sh [hunk 7]
```diff
@@ -1,4 +1,6 @@
 #!/bin/sh
```
Comment (sbdchd) [inline frontend/bootstrap.sh]: `#!/usr/bin/env sh`

File: frontend/bootstrap.sh [hunk 8]
```diff
@@ -1,4 +1,6 @@
...
+set -eux
```
Comment (sbdchd) [inline frontend/bootstrap.sh]: `#!/usr/bin/env sh`

File: maintenance_mode [hunk 9]
```diff
@@ -1,4 +1,6 @@
 #!/bin/sh
```
Comment (sbdchd) [inline maintenance_mode]: `#!/usr/bin/env sh`

File: upload [hunk 10]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
```
Comment (sbdchd) [inline upload]: `#!/usr/bin/env sh`

File: upload [hunk 11]
```diff
@@ -1,10 +1,9 @@
 #!/usr/bin/env sh
-set -eux
+set -eu
 
-docker tag recipemanager_nginx:latest cdignam/recipemanager_nginx:latest
-docker tag recipemanager_django:latest cdignam/recipemanager_django:latest
-docker tag recipemanager_react:latest cdignam/recipemanager_react:latest
+# https://github.com/moby/moby/issues/7336#issuecomment-194310594
+function dockerpush_alltags() {
+    for i in `docker images | grep ^recipeyak/ | grep -v '<none>' | awk 'BEGIN{OFS=":"}{print $1,$2}'`; do docker push $i && echo $i; done
```
Comment (sbdchd) [inline upload]: multiple lines please.

Also `$()` instead of ``

File: build [hunk 12]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env sh
+set -eux
```
Comment (sbdchd) [inline build]: Do we want `-eux` instead of `-e`?

File: deploy [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+# set tag to argument if provided
+tag="latest"
+if [ -n "$1" ]; then
+    tag=$1
+fi
+
+echo "Deploying tag: $tag"
+
+# pull new files
+docker pull cdignam/recipemanager_nginx:${tag}
```
Comment (sbdchd) [inline deploy]: I thought  we were using the `recipeyak` namespace now.

File: build (hunk 12 above)
Comment (chdsbd) [reply build]: Sure, it tells use what's going on and prevents unset variables

File: deploy (hunk 13 above)
Comment (chdsbd) [reply deploy]: I'll fix it.

**Pull request merged**